cats/cats-sql/src/main/kotlin/com/netflix/spinnaker/cats/sql/SqlProviderCache.kt

```diff
@@ -72,7 +72,7 @@ class SqlProviderCache(private val backingStore: WriteableCache) : ProviderCache
 
   override fun getAll(
     type: String,
-    identifiers: MutableCollection<String>?,
+    identifiers: MutableCollection<String?>?,
```

Comment:
We're still having issues with NPEs when using the V2 cache. What was the reason we needed to change these to nullable types?

Reply:
To prevent the [NPEs we saw in the first PR](https://github.com/spinnaker/clouddriver/pull/4755#issuecomment-664697250) since Java code interoping with Kotlin was adding null into a Kotlin MutableCollection<String> so it should be MutableCollection<String?>.

Reply:
The null identifiers weren't an issue before because we were querying for null identifiers, which returns no result. Now we need to filter them out since we have to hash the identifiers before any queries.

Reply:
_Something_ has regressed due to this PR, because the NPE issue persists, and it happens for all AWS and Titus operations. I'll pull this PR down and do some stepping.